gradle/micronaut/src/main/kotlin/com/equisoft/standards/gradle/micronaut/MicronautPlugin.kt

```diff
@@ -0,0 +1,141 @@
+import io.micronaut.gradle.docker.NativeImageDockerfile
+import org.graalvm.buildtools.gradle.dsl.GraalVMExtension
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.internal.provider.DefaultProvider
+import org.gradle.api.tasks.JavaExec
+import org.gradle.jvm.toolchain.JavaLanguageVersion
+import org.gradle.jvm.toolchain.JavaToolchainService
+import org.gradle.kotlin.dsl.configure
+import org.gradle.kotlin.dsl.create
+import org.gradle.kotlin.dsl.invoke
+import org.gradle.kotlin.dsl.withType
+import org.jetbrains.kotlin.allopen.gradle.AllOpenExtension
+import org.jetbrains.kotlin.allopen.gradle.AllOpenGradleSubplugin
+
+class MicronautPlugin : Plugin<Project> {
+    override fun apply(project: Project): Unit = with(project) {
+        val micronautSettingsExtension = extensions.create<MicronautSettingsExtension>("micronautSettings")
+
+        plugins.apply(MicronautAotPlugin::class.java)
+        plugins.apply(MicronautApplicationPlugin::class.java)
+        plugins.apply(AllOpenGradleSubplugin::class.java)
+
+        configureExtensions(micronautSettingsExtension)
+        configureTasks(project, micronautSettingsExtension)
+    }
+
+    private fun Project.configureExtensions(micronautSettingsExtension: MicronautSettingsExtension) {
+        configureAllOpen()
+        configureMicronautExtensionDefaultValues(micronautSettingsExtension)
+        configureDockerNativeImageDefaultValues(micronautSettingsExtension)
+    }
+
+    @SuppressWarnings("LongMethod", "MaxLineLength")
+    private fun Project.configureTasks(
+        project: Project,
+        micronautSettingsExtension: MicronautSettingsExtension
+    ) {
+        tasks {
+            fun configureDockerfile(dockerfile: Dockerfile) = with(dockerfile) {
+                val applicationVersion = project.version.toString()
+                environmentVariable("APPLICATION_VERSION", applicationVersion)
+                label(mapOf("org.opencontainers.image.version" to applicationVersion))
+            }
+
+            withType<MicronautDockerfile> {
+                baseImage.set(
+                    micronautSettingsExtension.javaVersion.map {
+                        "eclipse-temurin:${micronautSettingsExtension.javaVersion.get()}-jre"
+                    }
+                )
+                runCommand("useradd app && chown app -R /home/app")
+                user("app")
+                instruction("""HEALTHCHECK --interval=10s --timeout=3s --start-period=10s CMD curl --fail http://127.0.0.1:8081/health | grep '"status":"UP"' """) // ktlint-disable max-line-length
+                exposedPorts.set(micronautSettingsExtension.exposedPorts)
+                configureDockerfile(this)
+            }
+
+            withType<NativeImageDockerfile> {
+                baseImage.set(micronautSettingsExtension.runtimeImage)
+                exposedPorts.set(micronautSettingsExtension.exposedPorts)
+                configureDockerfile(this@withType)
```

Comment:
Juste `this` devrait être bon. Ou ajouter le `@withType` pour être cohérent avec L66

Reply:
C'est un vestige d'un autre essai